File: src/main/java/com/app_pedidos/model/entity/ItensPedido.java
package com.app_pedidos.model.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class ItensPedido {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private int quantidadeProduto;
    @OneToOne
    @JoinColumn(name = "produto_id")
    private Produto produto;

}

File: src/main/resources/templates/src/app/model/itensPedido.ts
import { Endereco } from "./endereco";

export class ItensPedido{
    id: number;
    produto_id: number;
    quantidadeProduto: number;

    constructor(id: number, produto_id: number, quantidadeProduto: number, public endereco?: Endereco){
        this.id = id;
        this.produto_id = produto_id;
        this.quantidadeProduto = quantidadeProduto;
        this.endereco = endereco;
    }
}